pkg/taints/taints.go

```
@@ -0,0 +1,135 @@
+
+// InitOutOfServiceTaintFlagsWithRetry tries to initialize the OutOfService taint info based on k8s version, in case it fails (potentially due to network issues) it will retry for a limited number of times.
+func InitOutOfServiceTaintFlagsWithRetry(ctx context.Context, config *rest.Config) error {
+	var err error
+	interval := 2 * time.Second // retry every 2 seconds
+	timeout := 10 * time.Second // for a period of 10 seconds
+
+	// Since the last internal error returned by InitOutOfServiceTaintFlags also indicates whether polling succeed or not, there is no need to also keep the context error returned by PollUntilContextTimeout.
+	// Using wait.PollUntilContextTimeout to retry initOutOfServiceTaintFlags in case there is a temporary network issue.
+	_ = wait.PollUntilContextTimeout(ctx, interval, timeout, true, func(ctx context.Context) (bool, error) {
+		if err = initOutOfServiceTaintFlags(config); err != nil {
+			return false, nil // Keep retrying
```

golangci-lint (2.11.4)

[error] error is not nil (line 92) but it returns nil

(nilerr)